data_labeling_examples/src/main/java/com/oracle/datalabelingservicesamples/scripts/CustomBulkLabelingScript.java

```diff
@@ -0,0 +1,177 @@
+			assert response.get__httpStatusCode__() == 200 : "List Record Response Error";
+
+			if (response.getRecordCollection().getItems().size() > 0) {
+				List<CompletableFuture<Void>> completableFutures = new ArrayList<>();
+				for (RecordSummary record : response.getRecordCollection().getItems()) {
+					List<String> label = Config.INSTANCE.getLabelingStrategy().getLabel(record);
+					if (null != label) {
+						CompletableFuture<Void> future = CompletableFuture
+								.runAsync(() -> processAnnotationForRecord(record, label), executorService);
+						completableFutures.add(future);
+					} else {
+						failedRecordIds.add(record.getId());
+					}
+				}
+				CompletableFuture<Void> combinedFuture = CompletableFuture
+						.allOf(completableFutures.toArray(new CompletableFuture[0]));
+				combinedFuture.get();
+			}
+
+			if (null == response.getOpcNextPage()) {
+				isUnlabeledRecordAvailable = false;
+			} else {
+				page = response.getOpcNextPage();
+			}
+
+		}
+
+		executorService.shutdown();
+		log.info("Time Taken for datasetId {}", datasetId);
+		log.info("Successfully Annotated {} record Ids", successRecordIds.size());
+		log.info("Failed record Ids {}", failedRecordIds);
+		long elapsedTime = System.nanoTime() - startTime;
+		log.info("Time Taken for datasetId {} is {} seconds", datasetId, elapsedTime/ 1_000_000_000);
+	}
+
+	private static void processAnnotationForRecord(RecordSummary record, List<String> labels) {
+		try {
+			List<Label> annotationLabel = new ArrayList<>();
+			for (String label : labels) {
+				annotationLabel.add(Label.builder().label(label).build());
+			}
+			List<Entity> entities = new ArrayList<>(
+					Collections.singleton(GenericEntity.builder().labels(annotationLabel).build()));
+
+			CreateAnnotationDetails annotationDetails = CreateAnnotationDetails.builder()
+					.compartmentId(record.getCompartmentId()).recordId(record.getId()).entities(entities).build();
+			CreateAnnotationRequest annotationReq = CreateAnnotationRequest.builder()
+					.createAnnotationDetails(annotationDetails).build();
+			Config.INSTANCE.getDlsDpClient().createAnnotation(annotationReq);
+			successRecordIds.add(record.getId());
+		} catch (Exception e) {
+			log.error("Exception in creating annotation for record {}", record);
+			failedRecordIds.add(record.getId());
+		}
+	}
+
+	private static void validateRequest(String datasetId, Map<String, List<String>> bulkLabelingRequest) {
+		/*
+		 * Validate Dataset
+		 */
+		GetDatasetRequest getDatasetRequest = GetDatasetRequest.builder().datasetId(datasetId).build();
+		GetDatasetResponse datasetResponse = Config.INSTANCE.getDlsDpClient().getDataset(getDatasetRequest);
+		assert datasetResponse.get__httpStatusCode__() == 200 : "Invalid Dataset Id Provided";
+		List<LabelName> datasetLabelSet = datasetResponse.getDataset().getLabelSet().getItems();
+
+		/*
+		 * Validate Request
+		 */
+		if (bulkLabelingRequest.size() > 3) {
+			log.error("Atmost 3 paths are allowed");
```

Comment:
Minor suggestion: Can we add bulkLabelingRequest.size() in log message to add more details? Or, else can we rephrase the error message as 'More than allowed limit of 3 paths were provided'.

Reply:
Updated.